packages/admin-api/src/controllers/application.controller.ts

```diff
@@ -22,8 +22,10 @@ export const getAllApplications = async (req: any, res: express.Response) => {
         const catchments = await getCatchments(req.kauth.grant.access_token)
         if (
             catchments.length === 0 ||
-            !filter.catchmentno ||
-            (filter.catchmentno !== -1 && !catchments.includes(filter.catchmentno))
+            filter.catchmentno == null ||
+            (Number(filter.catchmentno) !== -1 &&
+                Number(filter.catchmentno) !== 0 &&
+                !catchments.includes(filter.catchmentno))
```

Comment:
Does not restrict all-catchment search to IDIR users. Backend needs to restrict queries where catchment equals zero to IDIR users.